provider/http.go

```diff
@@ -14,15 +14,15 @@ import (
 type HTTPClient struct {
 	pb.UnimplementedApiServer
 
-	baseURL    string
-	httpClient *http.Client
-	requestID  utils.RequestID
-	privateKey *solana.PrivateKey
-	authHeader string
+	BaseURL    string
```

Comment:
reasoning behind this? I don't see a need to expose any of these attributes outside of this package based on this PR

Reply:
ah, for integration tests? Does the constructor now provide good enough configurability?